File: January/cpp/Daily Temperatures .cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        vector<int> ft(temperatures.size(),0);
        deque<int> dt;
        for(int i= temperatures.size()-1;i>=0;i--){
            if(dt.empty()){
                dt.push_front(i);
                ft[i] =0;
            }else{
                while(!dt.empty() && temperatures[i]>=temperatures[dt.front()]){
                    dt.pop_front();
                }
                if(dt.empty()){
                    ft[i] =0;
                }
                else{
                    ft[i] = dt.front()-i;
                }
                dt.push_front(i);
            }
        }
        return ft;
    }
};

File: January/Java/Daily Temperatures.java
import java.util.Stack;

public class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int n = temperatures.length;
        int[] result = new int[n];
        Stack<Integer> stack = new Stack<>();

        for (int i = n - 1; i >= 0; --i) {
            while (!stack.isEmpty() && temperatures[stack.peek()] <= temperatures[i]) {
                stack.pop();
            }

            if (!stack.isEmpty()) {
                result[i] = stack.peek() - i;
            }

            stack.push(i);
        }

        return result;
    }
}
